azext_edge/edge/providers/check/lnm.py

```diff
@@ -302,13 +314,24 @@ def _process_lnm_pods(
     conditions: Optional[List[str]] = None,
     namespace: Optional[str] = None,
     detail_level: int = ResourceOutputDetailLevel.summary.value,
+    resource_name: str = None,
 ) -> None:
     def _get_lnm_pods_namespace(pod: V1Pod) -> str:
         return pod.metadata.namespace
 
     pods = get_namespaced_pods_by_prefix(prefix=prefix, namespace=namespace, label_selector=label_selector)
 
+    if resource_name:
+        pods = filter_resources_by_name(resources=pods, resource_name=resource_name)
+
+        if not pods:
+            check_manager.add_target(target_name=target)
+            check_manager.add_display(target_name=target, display=Padding("Unable to fetch pods.", (0, 0, 0, padding + 2)))
+
+            return pods
```

Comment:
If the return type is `None` - no need to return a value.
Even if it looks like it's `None`, it could be `[]`, we can just use `return` to ensure no value is returned.

Edit: it looks like further down this method is also returning `pods` - is the return type incorrect?
Should we get the pods first and then pass them into this method to process them, or is there a reason it's fetching them and returning them?

Reply:
thanks for the good catch! there is no need to return for anything😋